cpp/src/text/subword/subword_tokenize.cu

```diff
@@ -165,9 +165,11 @@ tokenizer_result subword_tokenize(cudf::strings_column_view const& strings,
 {
   CUDF_EXPECTS(stride <= max_sequence_length,
                "stride must be less than or equal to max_sequence_length");
-  CUDF_EXPECTS(max_sequence_length * max_rows_tensor <
-                 static_cast<std::size_t>(std::numeric_limits<cudf::size_type>::max()),
-               "max_sequence_length x max_rows_tensor is too large for cudf output column size");
+  CUDF_EXPECTS(
+    max_sequence_length <=
+      (static_cast<uint64_t>(std::numeric_limits<cudf::size_type>::max()) / max_rows_tensor),
```

Comment:
Why use `uint64_t` here instead of `std::size_t` for both operands?

Reply:
I suppose `size_t == uint64_t` so why would there be a preference to use `size_t` here?
Knowing that `max_sequence_length` and `max_rows_tensor` are both `uint32_t` seemed more natural to use `uint64_t`.

Reply:
My preference is to use `std::size_t` because, though equivalent to `uint64_t` in most places, it has two benefits:
1. We know it's safe by definition (https://en.cppreference.com/w/cpp/types/size_t)
2. We can use the same type everywhere rather than `std::size_t` in some places and `uint64_t` in others. A common spelling in all overflow checks in libcudf, ideally defined by a detail overflow-checking function as mentioned in my PR comment, would be my preferred solution.